packages/field-base/src/shadow-focus-mixin.js

```diff
@@ -61,15 +58,13 @@ export const ShadowFocusMixin = (superClass) =>
       if (!this.disabled && this.focusElement) {
         const path = event.composedPath();
 
-        // When focus moves from outside and not with Shift + Tab, delegate it to focusElement.
+        // When focus moves to the host element itself, then delegate it to the focusElement
         // This should only move focus when using keyboard navigation, for clicks we don't want to interfere,
         // for example when the user tries to select some text
-        if (this._keyboardActive && path[0] === this && !this.contains(event.relatedTarget) && !this._isShiftTabbing) {
+        if (path[0] === this && this._keyboardActive) {
```

Comment:
I would suggest adding a test that verified that tabbing with Shift from `focusElement` doesn't move focus to the host element and consequently to `focusElement` as it might result in not being possible to escape the element when tabbing back.

Reply:
Very good point, that was not properly tested before. Done: https://github.com/vaadin/web-components/pull/3577/commits/72149564863095fc6935eaed4c13401b50f4d6da

Reply:
Ok, I see it is done. 👍